### PR TITLE
[DRAFT] `skaffold fix` to preserve comments.

### PR DESCRIPTION
**Pull request by tejal29**
fixes: #3955 
shdMerge after: #4201
**Description**
Use `yaml.Node` to Marshall old `Pipeline` and copy comments over into new upgraded `latest.Pipeline`

I have an PR to upgrade to yaml3. This PR is dependent of that.

**User facing changes (remove if N/A)**

**Follow-up Work (remove if N/A)**

### CODE REVIEW

File: cmd/skaffold/app/cmd/fix.go [hunk 1]
```diff
@@ -17,19 +17,23 @@ limitations under the License.
...
 	"context"
 	"fmt"
 	"io"
 	"io/ioutil"
 
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
-	yaml "gopkg.in/yaml.v2"
+	yaml2 "gopkg.in/yaml.v2"
+	yaml3 "gopkg.in/yaml.v3"
```
[the remark this answers is not in the document]
Comment (tejal29) [reply cmd/skaffold/app/cmd/fix.go:29]: @tstromberg I have a PR #4201 (mentioned in the description) to do the same

File: cmd/skaffold/app/cmd/fix.go [hunk 2]
```diff
@@ -17,19 +17,23 @@ limitations under the License.
...
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/color"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/util"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation"
+	misc "github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
```
[the remark this answers is not in the document]
Comment (tejal29) [reply cmd/skaffold/app/cmd/fix.go:36]: :). yeah we have a bunch of utils :)

File: cmd/skaffold/app/cmd/fix.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tejal29) [reply cmd/skaffold/app/cmd/fix.go:29]: This PR should be merged after #4201

**Comment by dgageot** [comment]
@tejal29 any plan on finishing this PR?

**Comment by tejal29** [comment]
I will resume this in next hackweek.

**Comment by tejal29** [comment]
closing again since its not a priority.

**Pull request closed without merging**